### PR TITLE
chore: sample app clean up and using datapipeline sdk

### PR DESCRIPTION
**Pull request by ami-aman**
linked to: https://linear.app/customerio/issue/MBL-71/sample-app-cleanup-with-cdp-sdk-apn-uikit

Includes:
- Initialisation of cdp sdk
- data model updates
- default storage update
- NSE update

Complete each step to get your pull request merged in. [Learn more about the workflow this project uses](https://github.com/customerio/customerio-ios/blob/develop/docs/dev-notes/GIT-WORKFLOW.md). 
- [ ] [Assign members of your team](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to review the pull request. 
- [ ] Wait for pull request status checks to complete. If there are problems, fix them until you see that [all status checks are passing](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fsymfony.com%2Fdoc%2F4.3%2F_images%2Fdocs-pull-request-symfonycloud.png&f=1&nofb=1). 
- [ ] Wait until the pull request has been reviewed *and approved* by a teammate
- [ ] After code reviews are approved and you determine this PR is ready to merge, select *Squash and Merge* button on this screen, leave the title and description to the default values, then merge the PR.

### CODE REVIEW

File: Apps/APN-UIKit/NotificationServiceExtension/NotificationService.swift [hunk 1]
```diff
@@ -7,10 +7,6 @@ class NotificationService: UNNotificationServiceExtension {
     var bestAttemptContent: UNMutableNotificationContent?
 
     override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
-        CustomerIO.initialize(siteId: BuildEnvironment.CustomerIO.siteId, apiKey: BuildEnvironment.CustomerIO.apiKey, region: Region.US) { config in
-            config.autoTrackPushEvents = true
-        }
```
Comment (levibostian) [inline Apps/APN-UIKit/NotificationServiceExtension/NotificationService.swift:12]: I know the CDP SDK changed the NSE public API from using `CustomerIO.initialize` to instead providing the siteid and apikey in the `MessagingPush` module. 

Are you planning on adding this in a future PR?

File: Apps/APN-UIKit/APN UIKit/Storage/UserDefaultKeys.swift [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import Foundation
 
 enum UserDefaultKeys: String {
-    case trackUrl
+    case cdnHost
```
Comment (levibostian) [inline Apps/APN-UIKit/APN UIKit/Storage/UserDefaultKeys.swift:4]: Do we need to add cdn host to the sample app? Are we planning on ever changing it during QA testing?

**Review by levibostian — APPROVED** [review]

File: Apps/APN-UIKit/APN UIKit/Model/Settings.swift [hunk 3]
```diff
@@ -2,9 +2,10 @@ import Foundation
 
 struct Settings {
     var deviceToken: String
-    var trackUrl: String
+    var cdnHost: String
+    var apiHost: String
     var siteId: String
-    var apiKey: String
+    var cdpWriteKey: String
```
Comment (Shahroz16) [inline Apps/APN-UIKit/APN UIKit/Model/Settings.swift:8]: why not just `writeKey`? 😅

**Review by Shahroz16 — APPROVED** [review]

**Review by mrehan27 — APPROVED** [review]

File: Apps/APN-UIKit/APN UIKit/Storage/UserDefaultKeys.swift (hunk 2 above)
Comment (ami-aman) [reply Apps/APN-UIKit/APN UIKit/Storage/UserDefaultKeys.swift:4]: TrackUrl has been _kind of_ broken down into two - cdnHost, apiKey. It is not clear if we plan to change these values during QA testing but I added these two fields as a replacement to trackUrl.

File: Apps/APN-UIKit/APN UIKit/Model/Settings.swift (hunk 3 above)
Comment (ami-aman) [reply Apps/APN-UIKit/APN UIKit/Model/Settings.swift:8]: To make it specific that this writeKey is for CDP. 😅 Does `cdpWriteKey` sound odd?

File: Apps/APN-UIKit/APN UIKit/AppDelegate.swift [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import CioDataPipelines
 import CioMessagingInApp
 import CioMessagingPushAPN
 import CioTracking
```
[the remark this answers is not in the document]
Comment (ami-aman) [reply Apps/APN-UIKit/APN UIKit/AppDelegate.swift:4]: This import is required for [AutoTrackingScreenViews](https://github.com/customerio/customerio-ios/blob/7e139dcf55c9bb0fbc011e8e9e6d589c4b21dece/Apps/APN-UIKit/APN%20UIKit/AppDelegate.swift#L57)

File: Apps/APN-UIKit/APN UIKit/AppDelegate.swift [hunk 5]
```diff
@@ -30,23 +31,30 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             storage.isTrackScreenEnabled = true
             storage.isTrackDeviceAttrEnabled = true
         }
+        var writeKey = BuildEnvironment.CustomerIO.cdpWriteKey
```
[the remark this answers is not in the document]
Comment (ami-aman) [reply Apps/APN-UIKit/APN UIKit/AppDelegate.swift:34]: Looks like most of you prefer `writeKey` over `cdpWriteKey`, hence I will update the property name.

[the remark this answers is not in the document]
Comment (ami-aman) [reply Apps/APN-UIKit/APN UIKit/AppDelegate.swift]: [Changes here](https://github.com/customerio/customerio-ios/pull/510/commits/6a83a73f72c73fdf55f860add4f376c4ab50c559)

File: Apps/APN-UIKit/APN UIKit/Model/Settings.swift (hunk 3 above)
Comment (ami-aman) [reply Apps/APN-UIKit/APN UIKit/Model/Settings.swift]: [Updated](https://github.com/customerio/customerio-ios/pull/510/commits/6a83a73f72c73fdf55f860add4f376c4ab50c559)

File: Apps/APN-UIKit/APN UIKit/AppDelegate.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (ami-aman) [reply Apps/APN-UIKit/APN UIKit/AppDelegate.swift]: [Updated](https://github.com/customerio/customerio-ios/pull/510/commits/7f5dd13713306e56aea6b968ca524fda7f71b043)

**Review by Shahroz16 — APPROVED** [review]

**Pull request merged**